### DISCUSSION TITLE
Download instead of opening

### DISCUSSION THREAD

**Discussion by Terribille** [DirectoryLister/DirectoryLister#899] [Help & Support] [answered]
Can DirectoryLister be configured to download files instead of opening them, all or specific?

**Comment by PHLAK** [comment, marked as answer]
Directory Lister does not handle the way files are opened directly. This is handled in combination by the web server and the user's web browser.

First and foremost ensure your server is configured to [return the correct MIME type](https://developer.mozilla.org/en-US/docs/Learn/Server-side/Configuring_server_MIME_types) of files.

  - For Apache this is accomplished with the [`AddType` directive](https://httpd.apache.org/docs/current/mod/mod_mime.html#addtype) (part of `mod_mime`)
  - For NGINX this is handled by the [`types` configuration option](http://nginx.org/en/docs/http/ngx_http_core_module.html#types) (usually found in the [`mime.types`](https://www.nginx.com/resources/wiki/start/topics/examples/full/#mime-types) file)

Additionally, the user's browser may be configured to handle all files of a given type in a specific fashion (i.e. open or download). Reference your browsers support page (e.g. [Firefox](https://support.mozilla.org/en-US/kb/change-firefox-behavior-when-open-file), [Chrome](https://support.google.com/chrome/answer/95759)) for instructions on configuring this.
